network/controller/bootstrap/bootstrap.go

```diff
@@ -182,16 +183,33 @@ func init() {
 
 // Bootstrap on the discovery node (step 1 of the bootstrap process)
 func (bc *bootstrapper) Bootstrap(ctx context.Context) (*network.BootstrapResult, *DiscoveryNode, error) {
-	log.Info("Bootstrapping to discovery node")
+	logger := inslogger.FromContext(ctx)
+	logger.Info("Bootstrapping to discovery node")
 	ctx, span := instracer.StartSpan(ctx, "Bootstrapper.Bootstrap")
 	defer span.End()
 	ch := bc.getDiscoveryNodesChannel(ctx, bc.Certificate.GetDiscoveryNodes(), 1)
-	result := bc.waitResultFromChannel(ctx, ch)
-	if result == nil {
-		return nil, nil, errors.New("Failed to bootstrap to any of discovery nodes")
+	discoveryCount := len(bc.Certificate.GetDiscoveryNodes())
+
+	var bootstrapResults []*network.BootstrapResult
+	var hosts []*host.Host
+	for {
+		bootstrapResults, hosts = bc.waitResultsFromChannel(ctx, ch, discoveryCount)
+		if len(bootstrapResults) == discoveryCount {
```

Comment:
If one of the discovery nodes is down, we will loop here until it is up again (maybe forever)

Reply:
it's copy-paste from discovery bootstrap and should be changed

Reply:
fixed